File: AbstractFactoryObg-c/AbstractFactory /AbstractFactory/AbstractFactoryAndFactorys/PoliceFactory.m
//
//  PoliceFactory.m
//  AbstractFactory
//
//  Created by Artem Esolnyak on 27.11.2019.
//  Copyright © 2019 Artem Esolnyak. All rights reserved.
//

#import "PoliceFactory.h"
#import "PoliceTruck.h"
#import "PoliceSUV.h"
#import "PoliceCar.h"

@implementation PoliceFactory

- (nonnull id<Car>)createCar {
    NSLog(@"ПОЛИЦЕЙСКАЯ ПАТРУЛЬНАЯ МАШИНА СОЗДАНА:");
    return [[PoliceCar alloc] initWithParameters];
    
    
}

- (nonnull id<SUVcar>)createSUvCar {
    NSLog(@"ПОЛИЦЕЙСКИЙ ВНЕДОРОЖНИК СОЗДАН:");
    return [[PoliceSUV alloc] initWithParameters];
}

- (nonnull id<Truck>)createTruck {
    NSLog(@"ПОЛИЦЕЙСКИЙ ГРУЗОВИК СОЗДАН:");
    return [[PoliceTruck alloc] initWithParameters];
}

@end

File: AbstractFabricSwift/AbstractFabric /AbstractFabric/AbstarctFactory/PoliceFactory.swift

import Foundation

class PoliceFactory : AbstarctFactory {
    
    func createTruck() -> Truck {
        print("Грузовик для полицейского управления создан")
        return PoliceTruck()
    }
    
    func createSuv() -> SUVcar {
        print("Внедорожник для полицейского управления создан")
        return PoliceSuv()
    }
    
    func createCar() -> Car {
        print("Патрульный автомобиль для полицейского управления создан")
        return PoliceCar() 
    }
    
    
}
